vault/logical_system.go

```diff
@@ -2602,11 +2604,20 @@ func (b *SystemBackend) handleEnableAuth(ctx context.Context, req *logical.Reque
 		Version:               pluginVersion,
 	}
 
-	resp, err := b.Core.handleDeprecatedMountEntry(ctx, me, consts.PluginTypeCredential)
-	if err != nil {
-		return handleError(err)
+	var resp *logical.Response
+
+	// Detect and handle deprecated auth methods
+	status, ok := b.Core.builtinRegistry.DeprecationStatus(me.Type, consts.PluginTypeCredential)
```

Comment:
I don't think this is using the same alias check that `handleDeprecatedMountEntry` is doing? And how come we call `DeprecationStatus` both here and inside the next function?

Reply:
Yeah, sorry, I don't think this was quite ready for review. I have some local updates which should clean a bunch of this cruft up.